src/shared/components/modal/createModalLauncher.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import { Modal, ModalProps } from '@patternfly/react-core';
+
+export type ModalComponentProps = Omit<ModalProps, 'children'>;
+
+export type CreateModalLauncher = <P extends ModalComponentProps>(
+  C: React.ComponentType<P>,
+) => (props: P) => (onClose: () => void) => React.ReactElement;
+
+export const createModalLauncher: CreateModalLauncher = (Component) => (props) => (onClose) => {
+  const handleClose = () => {
+    props.onClose && props.onClose();
+    onClose();
+  };
+  return (
+    <Modal
+      ref={null}
```

Comment:
Why is ref set to null here?

Reply:
This was to fix a type error, removed it and handled the error